taf/updater/handlers.py

```diff
@@ -288,40 +273,17 @@ def get_current_targets(self):
         except GitError:
             return []
 
-    def get_mirrors(self, _file_type, _file_path):
-        # pylint: disable=unused-argument
-        # return a list containing just the current commit
-        return [self.current_commit]
-
-    def get_metadata_file(self, file_mirror, file_name, _upperbound_filelength=None):
-        return self._get_file(file_mirror, "metadata/" + file_name)
-
-    def get_target_file(self, file_mirror, _file_length, _download_safely, file_path):
-        return self._get_file(file_mirror, "targets/" + file_path)
-
-    def _get_file(self, commit, filepath):
-        f = self.validation_auth_repo.get_file(commit, filepath)
-        temp_file_object = securesystemslib.util.TempFile()
-        temp_file_object.write(f.encode())
-        return temp_file_object
-
-    def get_file_digest(self, filepath, algorithm):
-        filepath = Path(filepath).relative_to(self.validation_auth_repo.get_file)
-        file_obj = self._get_file(self.current_commit, filepath)
-        return securesystemslib.hash.digest_fileobject(file_obj, algorithm=algorithm)
-
-    def on_successful_update(self, filename, mirror):
-        # after the is successfully completed, set the
-        # next commit as current for the given file
-        taf_logger.debug("{} updated from commit {}", filename, mirror)
-
-    def on_unsuccessful_update(self, filename):
-        taf_logger.error("Failed to update {}", filename)
+    def get_current_target_data(self, filepath, raw=False):
+        return self.validation_auth_repo.get_file(
+            self.current_commit, f"targets/{filepath}", raw=raw
+        )
 
     def update_done(self):
-        # the only metadata file that is always updated
-        # regardless of if it changed or not is timestamp
-        # so we can check if timestamp was updated a certain
-        # number of times
+        """Used to indicate whether updater has finished with update.
+        Update is considered done when all commits have been validated"""
         self.current_commit_index += 1
+
+        if self.current_commit_index == (len(self.commits) - 1):
```

Comment:
I'm not really sure where to put this other than here. We want to revert our TUF expiration validation patch for most recent (last) commit. Since we only need to revert this patch during the updater `while not git_updater.update_done():` code block and only on the last commit, moving this to cleanup won't be adequate. Maybe turn the check into a decorator above `update_done`??

Reply:
Does 4d6b12ff7e53fc948cd2449882c33627c6140010 look better?

Reply:
Yeah, that makes sense. `update_done` might already updates the current index, probably not the best name. Or we need to split that into two methods. Since an older issue unrelated to this PR, I'm fine with both the decorator and the old implementation, whatever you prefer